lib/phoenix_live_view/helpers.ex

```diff
@@ -863,27 +902,70 @@ defmodule Phoenix.LiveView.Helpers do
 
   ## Examples
 
-      <.form_for let={f} data={@changeset}>
+      <.form let={f} for={@changeset}>
         <%= text_input f, :name %>
-      </.form_for>
+      </.form>
 
-      <.form_for let={user_form} data={@changeset} as="user">
+      <.form let={user_form} for={@changeset} as="user" {@extra}>
         <%= text_input user_form, :name %>
-      </.form_for>
+      </.form>
   """
-  def form_for(assigns) do
-    opts =
-      assigns |> Map.take([:as, :method, :multipart, :csrf_token, :errors, :id]) |> Enum.into([])
+  @form_opts [:as, :method, :multipart, :csrf_token, :errors, :id]
+  def form(assigns) do
+    action = assigns[:action] || "#"
+
+    {form_for, assigns} =
+      Map.pop_lazy(assigns, :for, fn -> raise ArgumentError, "missing :for assign to form" end)
+
+    {csrf_token, assigns} =
+      Map.pop_lazy(assigns, :csrf_token, fn -> Phoenix.HTML.Tag.csrf_token_value(action) end)
+
+    form_opts = assigns |> Map.take(@form_opts) |> Enum.into([])
+
+    form = %Phoenix.HTML.Form{Phoenix.HTML.FormData.to_form(form_for, form_opts) | action: action}
```

Comment:
I would pass all options to `to_form`, because that's what `Phoenix.HTML` does. We would also pass `:__changed__` and other stuff, but I think that's fine.

```suggestion
    form_opts = Map.to_list(assigns)

    form = %Phoenix.HTML.Form{Phoenix.HTML.FormData.to_form(form_for, form_opts) | action: action}
```